lib/credentials/shared_ini_file_credentials.js

```diff
@@ -87,6 +100,75 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
   /**
    * @api private
    */
+  loadRoleProfile: function loadRoleProfile(creds, callback) {
```

Comment:
Done. The method still requires the `creds` since the `source_profile` has to be retrieved.